pkg/kubectl/cmd/cmd.go

```diff
@@ -303,6 +304,44 @@ func NewDefaultKubectlCommand() *cobra.Command {
 	return NewDefaultKubectlCommandWithArgs(NewDefaultPluginHandler(plugin.ValidPluginFilenamePrefixes), os.Args, os.Stdin, os.Stdout, os.Stderr)
 }
 
+func lookupFlag(flag string, fs *pflag.FlagSet) *pflag.Flag {
+	flag = strings.SplitN(flag, "=", 2)[0]
+	switch {
+	case strings.HasPrefix(flag, "--"):
+		return fs.Lookup(flag[2:])
+	case strings.HasPrefix(flag, "-"):
+		return fs.ShorthandLookup(flag[1:2])
```

Comment:
Is this OK?

My concern is if someone accidentally does `-` when they meant `--` and if this only takes the first character after the `-` then it could cause the wrong flag to be used instead of failing.

For example, in `kubectl run` there is an `--image` flag and a `-i` flag which is shorthand for `--stdin`.

In this case it probably would not cause a bad problem because the command would fail, but if I used `-image` instead of `--image`, I think the code above would interpret `-image` as `-i`, right?

What if it just does this?
```
return fs.ShorthandLookup(flag[1:])
```

Or return an error if the length of the single-dash flag is not 1.

Just an idea

Reply:
Thanks for catching that, I handled it in the original version and mistakenly removed it